service-registry/src/main/java/io/servicecomb/serviceregistry/ServiceRegistry.java

```diff
@@ -47,6 +49,8 @@
   List<MicroserviceInstance> findServiceInstance(String appId, String microserviceName,
       String microserviceVersionRule);
 
+  FindInstancesResponse findServiceInstance(InstanceCache instanceCache);
```

Comment:
do not use instanceCache as the input parameter
it's internal class we are planing to delete InstanceCache class.

return SC's response(FindInstancesResponse ) is not good, interface ServiceRegistry is designed for easy use, not for communicate with SC.(communicate interface is ServiceRegistryClient)

purpose of this method:
1.base on old findServiceInstance, add a new input parameter to indicate exists data version
2.if exists data is the latest data, then return xxx, otherwise return new data.